python/ray/data/preprocessors/batch_mapper.py

```diff
@@ -36,19 +39,53 @@ class BatchMapper(Preprocessor):
         >>> preprocessor = BatchMapper(fn)
         >>> preprocessor.transform(ds)  # doctest: +SKIP
         Dataset(num_blocks=1, num_rows=3, schema={X: int64})
+        >>>
+        >>> def fn_numpy(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
+        ...     return {"X": batch["X"]}
+        >>> preprocessor = BatchMapper(fn_numpy, batch_format="numpy")
+        >>> preprocessor.transform(ds)  # doctest: +SKIP
+        Dataset(num_blocks=1, num_rows=3, schema={X: int64})
 
     Args:
         fn: The function to apply to data batches.
+        batch_format: The preferred batch format to use in UDF. If not given,
+            default to pandas.
     """
 
     _is_fittable = False
 
-    def __init__(self, fn: Callable[["pandas.DataFrame"], "pandas.DataFrame"]):
+    def __init__(
+        self,
+        fn: Callable[
+            [Union["pandas.DataFrame", np.ndarray, Dict[str, np.ndarray]]],
+            Union["pandas.DataFrame", np.ndarray, Dict[str, np.ndarray]],
+        ],
+        batch_format="pandas",
+        # TODO: We should reach consistency of args between BatchMapper and map_batches.
+    ):
+        if batch_format not in [
+            "pandas",
+            "numpy",
+        ]:
+            raise ValueError("BatchMapper only supports pandas and numpy batch format.")
```

Comment:
Is None not allowed? Should be `if batch_format and batch_format not in ["pandas", "numpy"]`?

Reply:
i haven't seen a case that we got None here yet

Reply:
ah what I mean is there should there be an option that doesn't require the user to manually specify? If `batch_format` is None, then we can just use the transform that matches with the data format.